### ISSUE TITLE
Entire chat history visible when joining new (private) channel

### ISSUE THREAD

**Issue by stefuNz** [RocketChat/Rocket.Chat#1442]
When i add a new member to a channel, i don't want him to see everything that has been written.

Is this a bug or just not yet implemented?

**Comment by graywolf336** [comment]
This is a feature. But what I'm guessing you are expecting is something like how slack handles this, they ask you whether you want the new person to see the private group's history or not. If yes, then they are just added to the group. If no, then a new group is created with all the people in the room plus the new person being invited. Is this what you're expected?

**Comment by stefuNz** [comment]
Yes, exactly, this is what i expected. I want to choose that the new person can't see the history. Is that feature on the roadmap?

**Comment by ninja-** [comment]
it's on the roadmap at some point...I have to admit that slack's approach to this problem is the best possible one

**Comment by wtsarchive** [comment]
Is it possible to make that setting available for public channels as well? I would benefit from a setting where new users joining a channel don't see its history, only new messages from the time he joined the channel forward.

**Comment by graywolf336** [comment]
@wtsarchive I don't believe there's any plans for adding this to *public* channels, unless there becomes a big need for it or someone submits a pull request adding the feature. The other team members can correct me if I'm wrong though.

**Comment by geekgonecrazy** [comment]
What value does this really add?  Any situation i've been in where I didn't want someone else to be able to read past conversations, we just simply created a new group and brought the user into that group.

**Comment by wargamesqcf** [comment]
would love this feature.

**Comment by geekgonecrazy** [comment]
So the proposal here is... any message sent prior to a new person joining would be un-viewable by the new person?  I'm guessing you'd still want the people there prior to be able to see the history before they joined?

Also i'm guessing some people would want the new person to be able to see history... so we couldn't just do this by default.  We'd have to ask them each time if they wanted to allow the person invited to see history.

Also we'd then have to keep track of who could see each individual message and who couldn't...

**Comment by wargamesqcf** [comment]
or just obfuscate via join date?

**Comment by geekgonecrazy** [comment]
That could work.  I don't think we keep track of join date to a channel.  We post a message on join... Which we could use, but that would become an expensive operation.  So would have to store the join date for each person, as well as if they were restricted access to history or not.

**Comment by wargamesqcf** [comment]
having admins view the entire log is preferable to users being able to see everything, then again my use is not typical of what most users have in that I am running an tabletop roleplaying chat and don't need people spying on past conversations.

**Comment by geekgonecrazy** [comment]
Maybe we can have this configurable at the channel level.  So the channel owner or site admin can determine if a channel would allow users to view past messages on join.  I can see many other applications for this too.  But couldn't be a default behavior because most of our users expect it to work as it does now.

**Comment by wargamesqcf** [comment]
maybe also have it as a switch, all new created rooms have this feature auto enabled or disabled?

**Comment by johnyb0y** [comment]
Any Update on this feature request ? Thank you!

**Comment by bussyjd** [comment]
👍

**Comment by Nuvelle** [comment]
This is something we are looking for at the moment, without it we may not be able to use telegram at all :-/

**Comment by ManasN** [comment]
Any update? thanks

**Comment by geekgonecrazy** [comment]
I think this is related to #9106 If we had the ability to have group dm's we could likely prompt and do this when someone tried to add a new member to group dm

But private groups work exactly as designed right now.

**Comment by meghaParmar** [comment]
Is there any way to prevent old chat history from new member in group?

**Comment by GoetheG** [comment]
Hey guys. hey @geekgonecrazy: Are there any updates regarding this topic? I find it very important to let the administrator decide whether new users can or cannot read previous messages, before their date of entry in a channel.

**Comment by GoetheG** [comment]
Is there anthing new here? this is a serious compliance issue for me and my club. We cannot allow new users to see the chats there were in a channel before the user joined the channel.

**Comment by anuraag94** [comment]
Agreed, I'm a researcher using Slack as a platform to moderate small group interactions. To ensure the validity of my research, I need the history of all channels to be hidden from new users, including #general

**Comment by GoetheG** [comment]
@meghaParmar @anuraag94 @2betafactor @ManasN @johnyb0y @bussyjd @wargamesqcf and everybody else. Would you support a bounty on Bountysource for that with me?

**Comment by johnyb0y** [comment]
Hi @GoetheG To us it's really not that important. We've communicated this with our users and so far there have been no serious social incidents :)

**Comment by xmxax** [comment]
Hey there! I am also looking for this feature! 

Are there any updates or plans to actually implement this? 

Thank you for your hard work!

**Comment by Aens** [comment]
^ Let me know also if there's a way to do it.

**Comment by CaptWeiss** [comment]
I support this feature be added

**Comment by mrtnbm** [comment]
Yes, should be added

**Comment by suyon-won** [comment]
We can have a json['subscription']['ts'] from /api/v1/subscriptions.getOne.
And query /api/v1/groups.history with the 'oldest' option and ts.

**Comment by geekgonecrazy** [comment]
Can you try group dm?  If you add user here I believe creates new dm.  From what I can tell this still works exactly as we intend

**Comment by somoza** [comment]
+1

**Comment by dirksan28** [comment]
After reading this thread - I guess there is still no such feature implemented right now... right?
